SyncTests/RecordTests.swift

```diff
@@ -62,15 +63,18 @@ class RecordTests: XCTestCase {
 
         print(clientPayload, terminator: "\n")
 
-        // Only payloads that parse as JSON are valid.
-        XCTAssertNil(Record<CleartextPayloadJSON>.fromEnvelope(EnvelopeJSON(invalidPayload), payloadFactory: clearFactory))
+        // Non-JSON malformed payloads don't even yield a value.
+        XCTAssertNil(Record<CleartextPayloadJSON>.fromEnvelope(EnvelopeJSON(malformedPayload), payloadFactory: clearFactory))
+
+        // Only payloads that parse as JSON objects are valid.
+        XCTAssertFalse(Record<CleartextPayloadJSON>.fromEnvelope(EnvelopeJSON(invalidPayload), payloadFactory: clearFactory)!.payload.isValid())
 
         // Missing ID.
-        XCTAssertNil(Record<CleartextPayloadJSON>.fromEnvelope(EnvelopeJSON(emptyPayload), payloadFactory: clearFactory))
+        XCTAssertFalse(Record<CleartextPayloadJSON>.fromEnvelope(EnvelopeJSON(emptyPayload), payloadFactory: clearFactory)!.payload.isValid())
```

Comment:
I think it might also be handy here to do an `XCTAssertNotNil` on `Record<CleartextPayloadJSON>.fromEnvelope(EnvelopeJSON(emptyPayload), payloadFactory: clearFactory)` before the `XCTAssertFalse` here, otherwise if it does return nil (for some reason) it will crash the tests, which isn't brilliant.

Ditto for those below that also force unwrap